RwsCheck.py

```diff
@@ -170,6 +172,23 @@ def check_exclusivity(self, check_sets):
                 else:
                     site_list.update(aliases)
 
+    def check_associated_count(self, check_sets):
+        """This method checks for RwsSets that exceed the associated limit
+
+        Creates a warning for each set passed in the check_sets list that has 
+        more associatedSites than the ASSOCIATED_LIMIT
+
+        Args:
+            check_sets: Dict[string, RwsSet]
+        Returns:
```

Comment:
For future refactoring: IMO it'd be better design to make this a ["pure" function](https://en.wikipedia.org/wiki/Pure_function), i.e. it takes all its inputs as arguments, and returns its result explicitly (rather than just mutating the instance). That makes it easier to test, and easier for a reader to understand how to use the method.

Doing that in this CL would require the caller to know whether the results should be considered warnings or errors, though, which isn't great. What'd be nice is if each check returned a list of errors *and* and list of warnings, and the caller just appended each to its running lists. That's a larger change, so it shouldn't be part of this PR.